File: artifacts/heman/decompose/src/globals.c
extern float _gamma
/*** DEPENDENCIES:
***/


const float INF = 1E20
/*** DEPENDENCIES:
***/


static const float SEALEVEL = 0.5
/*** DEPENDENCIES:
***/


static const float DEFAULT_STRENGTH = 0.6
/*** DEPENDENCIES:
***/


static float _occlusion_scale = 1.0f
/*** DEPENDENCIES:
***/


static const int8_t gradients2D[] = {5, 2, 2, 5, -5, 2, -2, 5, 5, -2, 2, -5, -5, -2, -2, -5}
/*** DEPENDENCIES:
***/


static const signed char gradients3D[] = {-11, 4, 4, -4, 11, 4, -4, 4, 11, 11, 4, 4, 4, 11, 4, 4, 4, 11, -11, -4, 4, -4, -11, 4, -4, -4, 11, 11, -4, 4, 4, -11, 4, 4, -4, 11, -11, 4, -4, -4, 11, -4, -4, 4, -11, 11, 4, -4, 4, 11, -4, 4, 4, -11, -11, -4, -4, -4, -11, -4, -4, -4, -11, 11, -4, -4, 4, -11, -4, 4, -4, -11}
/*** DEPENDENCIES:
***/


static const signed char gradients4D[] = {3, 1, 1, 1, 1, 3, 1, 1, 1, 1, 3, 1, 1, 1, 1, 3, -3, 1, 1, 1, -1, 3, 1, 1, -1, 1, 3, 1, -1, 1, 1, 3, 3, -1, 1, 1, 1, -3, 1, 1, 1, -1, 3, 1, 1, -1, 1, 3, -3, -1, 1, 1, -1, -3, 1, 1, -1, -1, 3, 1, -1, -1, 1, 3, 3, 1, -1, 1, 1, 3, -1, 1, 1, 1, -3, 1, 1, 1, -1, 3, -3, 1, -1, 1, -1, 3, -1, 1, -1, 1, -3, 1, -1, 1, -1, 3, 3, -1, -1, 1, 1, -3, -1, 1, 1, -1, -3, 1, 1, -1, -1, 3, -3, -1, -1, 1, -1, -3, -1, 1, -1, -1, -3, 1, -1, -1, -1, 3, 3, 1, 1, -1, 1, 3, 1, -1, 1, 1, 3, -1, 1, 1, 1, -3, -3, 1, 1, -1, -1, 3, 1, -1, -1, 1, 3, -1, -1, 1, 1, -3, 3, -1, 1, -1, 1, -3, 1, -1, 1, -1, 3, -1, 1, -1, 1, -3, -3, -1, 1, -1, -1, -3, 1, -1, -1, -1, 3, -1, -1, -1, 1, -3, 3, 1, -1, -1, 1, 3, -1, -1, 1, 1, -3, -1, 1, 1, -1, -3, -3, 1, -1, -1, -1, 3, -1, -1, -1, 1, -3, -1, -1, 1, -1, -3, 3, -1, -1, -1, 1, -3, -1, -1, 1, -1, -3, -1, 1, -1, -1, -3, -3, -1, -1, -1, -1, -3, -1, -1, -1, -1, -3, -1, -1, -1, -1, -3}
/*** DEPENDENCIES:
***/


extern const kmVec3 KM_VEC3_NEG_Z
/*** DEPENDENCIES:
typedef struct kmVec3
{
  float x;
  float y;
  float z;
} kmVec3
----------------------------
***/


extern const kmVec3 KM_VEC3_POS_Z
/*** DEPENDENCIES:
typedef struct kmVec3
{
  float x;
  float y;
  float z;
} kmVec3
----------------------------
***/


extern const kmVec3 KM_VEC3_POS_Y
/*** DEPENDENCIES:
typedef struct kmVec3
{
  float x;
  float y;
  float z;
} kmVec3
----------------------------
***/


extern const kmVec3 KM_VEC3_NEG_Y
/*** DEPENDENCIES:
typedef struct kmVec3
{
  float x;
  float y;
  float z;
} kmVec3
----------------------------
***/


extern const kmVec3 KM_VEC3_NEG_X
/*** DEPENDENCIES:
typedef struct kmVec3
{
  float x;
  float y;
  float z;
} kmVec3
----------------------------
***/


extern const kmVec3 KM_VEC3_POS_X
/*** DEPENDENCIES:
typedef struct kmVec3
{
  float x;
  float y;
  float z;
} kmVec3
----------------------------
***/


extern const kmVec3 KM_VEC3_ZERO
/*** DEPENDENCIES:
typedef struct kmVec3
{
  float x;
  float y;
  float z;
} kmVec3
----------------------------
***/


extern const kmVec2 KM_VEC2_POS_Y
/*** DEPENDENCIES:
typedef struct kmVec2
{
  float x;
  float y;
} kmVec2
----------------------------
***/


extern const kmVec2 KM_VEC2_NEG_Y
/*** DEPENDENCIES:
typedef struct kmVec2
{
  float x;
  float y;
} kmVec2
----------------------------
***/


extern const kmVec2 KM_VEC2_NEG_X
/*** DEPENDENCIES:
typedef struct kmVec2
{
  float x;
  float y;
} kmVec2
----------------------------
***/


extern const kmVec2 KM_VEC2_POS_X
/*** DEPENDENCIES:
typedef struct kmVec2
{
  float x;
  float y;
} kmVec2
----------------------------
***/


extern const kmVec2 KM_VEC2_ZERO
/*** DEPENDENCIES:
typedef struct kmVec2
{
  float x;
  float y;
} kmVec2
----------------------------
***/

File: artifacts/heman/translate_without_break_down/src/globals.rs
use crate::*;
use lazy_static::lazy_static;


lazy_static! {
    pub static ref _GAMMA: std::sync::Mutex<f32> = std::sync::Mutex::new(0.0);
}

pub const INF: f32 = 1E20;

pub const SEALEVEL: f32 = 0.5;

pub const DEFAULT_STRENGTH: f32 = 0.6;


lazy_static! {
    pub static ref _OCCLUSION_SCALE: std::sync::Mutex<f32> = std::sync::Mutex::new(1.0f32);
}

pub static GRADIENTS2D: [i8; 16] = [
    5, 2,  // 2 elements
    2, 5,  // 2 elements
    -5, 2, // 2 elements
    -2, 5, // 2 elements
    5, -2, // 2 elements
    2, -5, // 2 elements
    -5, -2, // 2 elements
    -2, -5  // 2 elements
];

pub static GRADIENTS3D: [i8; 72] = [
    -11, 4, 4, -4, 11, 4, -4, 4, 11,  // 9 elements
    11, 4, 4, 4, 11, 4, 4, 4, 11,     // 9 elements
    -11, -4, 4, -4, -11, 4, -4, -4, 11, // 9 elements
    11, -4, 4, 4, -11, 4, 4, -4, 11,  // 9 elements
    -11, 4, -4, -4, 11, -4, -4, 4, -11, // 9 elements
    11, 4, -4, 4, 11, -4, 4, 4, -11,  // 9 elements
    -11, -4, -4, -4, -11, -4, -4, -4, -11, // 9 elements
    11, -4, -4, 4, -11, -4, 4, -4, -11, // 9 elements
];

pub static GRADIENTS4D: [i8; 256] = [
    3, 1, 1, 1, 1, 3, 1, 1, 1, 1, 3, 1, 1, 1, 1, 3, 
    -3, 1, 1, 1, -1, 3, 1, 1, -1, 1, 3, 1, -1, 1, 1, 3, 
    3, -1, 1, 1, 1, -3, 1, 1, 1, -1, 3, 1, 1, -1, 1, 3, 
    -3, -1, 1, 1, -1, -3, 1, 1, -1, -1, 3, 1, -1, -1, 1, 3, 
    3, 1, -1, 1, 1, 3, -1, 1, 1, 1, -3, 1, 1, 1, -1, 3, 
    -3, 1, -1, 1, -1, 3, -1, 1, -1, 1, -3, 1, -1, 1, -1, 3, 
    3, -1, -1, 1, 1, -3, -1, 1, 1, -1, -3, 1, 1, -1, -1, 3, 
    -3, -1, -1, 1, -1, -3, -1, 1, -1, -1, -3, 1, -1, -1, -1, 3, 
    3, 1, 1, -1, 1, 3, 1, -1, 1, 1, 3, -1, 1, 1, 1, -3, 
    -3, 1, 1, -1, -1, 3, 1, -1, -1, 1, 3, -1, -1, 1, 1, -3, 
    3, -1, 1, -1, 1, -3, 1, -1, 1, -1, 3, -1, 1, -1, 1, -3, 
    -3, -1, 1, -1, -1, -3, 1, -1, -1, -1, 3, -1, -1, -1, 1, -3, 
    3, 1, -1, -1, 1, 3, -1, -1, 1, 1, -3, -1, 1, 1, -1, -3, 
    -3, 1, -1, -1, -1, 3, -1, -1, -1, 1, -3, -1, -1, 1, -1, -3, 
    3, -1, -1, -1, 1, -3, -1, -1, 1, -1, -3, -1, 1, -1, -1, -3, 
    -3, -1, -1, -1, -1, -3, -1, -1, -1, -1, -3, -1, -1, -1, -1, -3
];


lazy_static! {
    pub static ref KM_VEC3_NEG_Z: KmVec3 = KmVec3 {
        x: 0.0,
        y: 0.0,
        z: -1.0,
    };
}


lazy_static! {
    pub static ref KM_VEC3_POS_Z: KmVec3 = KmVec3 {
        x: 0.0,
        y: 0.0,
        z: 1.0
    };
}


lazy_static! {
    pub static ref KM_VEC3_POS_Y: KmVec3 = KmVec3 {
        x: 0.0,
        y: 1.0,
        z: 0.0,
    };
}


lazy_static! {
    pub static ref KM_VEC3_NEG_Y: KmVec3 = KmVec3 {
        x: 0.0,
        y: -1.0,
        z: 0.0,
    };
}


lazy_static! {
    pub static ref KM_VEC3_NEG_X: KmVec3 = KmVec3 {
        x: -1.0,
        y: 0.0,
        z: 0.0,
    };
}


lazy_static! {
    pub static ref KM_VEC3_POS_X: KmVec3 = KmVec3 {
        x: 1.0,
        y: 0.0,
        z: 0.0,
    };
}


lazy_static! {
    pub static ref KM_VEC3_ZERO: KmVec3 = KmVec3 {
        x: 0.0,
        y: 0.0,
        z: 0.0,
    };
}


lazy_static! {
    pub static ref KM_VEC2_POS_Y: KmVec2 = KmVec2 { x: 0.0, y: 1.0 };
}


lazy_static! {
    pub static ref KM_VEC2_NEG_Y: KmVec2 = KmVec2 { x: 0.0, y: -1.0 };
}


lazy_static! {
    pub static ref KM_VEC2_NEG_X: KmVec2 = KmVec2 { x: -1.0, y: 0.0 };
}


lazy_static! {
    pub static ref KM_VEC2_POS_X: KmVec2 = KmVec2 { x: 1.0, y: 0.0 };
}


lazy_static! {
    pub static ref KM_VEC2_ZERO: KmVec2 = KmVec2 { x: 0.0, y: 0.0 };
}

